scripts/003-gdb-PPU.sh

```diff
@@ -3,41 +3,31 @@
 
 GDB="gdb-7.5.1"
 
-if [ ! -d ${GDB} ]; then
-
-  ## Download the source code.
-  if [ ! -f ${GDB}.tar.bz2 ]; then wget --continue https://ftp.gnu.org/gnu/gdb/${GDB}.tar.bz2; fi
-
-  ## Download an up-to-date config.guess and config.sub
-  if [ ! -f config.guess ]; then wget --continue https://git.savannah.gnu.org/cgit/config.git/plain/config.guess; fi
-  if [ ! -f config.sub ]; then wget --continue https://git.savannah.gnu.org/cgit/config.git/plain/config.sub; fi
-
-  ## Unpack the source code.
-  tar xfvj ${GDB}.tar.bz2
-
-  ## Replace config.guess and config.sub
-  cp config.guess config.sub ${GDB}
-
+PS3DEV_SOURCES=`git rev-parse --show-toplevel`/src
+BUILD_HOST=`${PS3DEV_SOURCES}/config/config.guess`
+PROC=$(nproc --all 2>&1)
+if [ -n ${PROC-} ]; then
+  PROC=4;
 fi
 
-if [ ! -d ${GDB}/build-ppu ]; then
-
-  ## Create the build directory.
-  mkdir ${GDB}/build-ppu
+git submodule update --init --remote --recursive ${PS3DEV_SOURCES}/${GDB}
+git submodule update --init --remote --recursive ${PS3DEV_SOURCES}/config
```

Comment:
The `--remote` flag makes your working tree dirty, so it should probably not be used automatically.  It would also be nice if you can build without network access if you have everything cloned already.

Reply:
Should I find a way to use --remote if network is present and omit if not?
We sort of want to get the updates even if they make the submodule dirty, maybe we can git submodule foreach git clean -df after?

Reply:
I think I'd prefer the other solution you suggested, that a push generates a bump commit automatically.
If we are to commit the version pin bumps manually then we should also run `update --remote` manually first.

Reply:
And automatically doing `git clean` is a terrible idea and also does not solve the problem at hand:  `update --remote` makes the working tree dirty by modifying existing objects (namely the version pin), not by adding untracked files.

Reply:
Just going to keep this branch updated for awhile while we get all the repos in line with the same tactic and work on workflows. No need to rush merging this.

Reply:
Yeah, and there's no problem syncing manually to begin with; it's no more work than manually updating patches, just less error prone.  :smile: